keanu-project/src/main/java/io/improbable/keanu/tensor/NumberTensor.java

```diff
@@ -77,6 +79,16 @@
     T setWithMask(T mask, N value);
 
     T apply(Function<N, N> function);
+
+    default List<T> sliceAlongDimension(int dimension, long indexStart, long indexEnd) {
+        List<T> slicedTensors = new ArrayList<>();
```

Comment:
I don't think this is necessary. See below, you can achieve what you're doing here by using `split`.